ingest.go

```diff
@@ -314,33 +318,34 @@ func ingestLink(
 		}
 	}
 
-	return nil
+	return newPaths, nil
 }
 
 func ingestMemtableOverlaps(cmp Compare, mem flushable, meta []*fileMetadata) bool {
 	iter := mem.newIter(nil)
 	rangeDelIter := mem.newRangeDelIter(nil)
 	rkeyIter := mem.newRangeKeyIter(nil)
 
+	closeIters := func() error {
+		err := iter.Close()
+		if rangeDelIter != nil {
+			err = firstError(err, rangeDelIter.Close())
+		}
+		if rkeyIter != nil {
+			err = firstError(err, rkeyIter.Close())
+		}
+		return err
+	}
+
 	for _, m := range meta {
 		if overlapWithIterator(iter, &rangeDelIter, rkeyIter, m, cmp) {
+			closeIters()
```

Comment:
I guess this iterator leak was not too bad because it doesn't hold a ref and will eventually be GC'd. Though I'm glad you found it.